### PR TITLE
string: strerror_r() implementation

### PR DESCRIPTION
**Pull request by agkaminski**
DONE: RTOS-554

## Description

Fixes https://github.com/phoenix-rtos/phoenix-rtos-project/issues/694

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: (ia32-generic-qemu).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

**Review by mateusz-bloch — COMMENTED** [review]
LGTM, checked strerror_r on emu and hardware targets

**Review by badochov — CHANGES_REQUESTED** [review]
`strerror` should only have race with itself, but in current implementation there's a race between `strerror` and `gai_strerror`, I'd make `err_unknownMsg`  a static variable inside each of this functions

File: string/strerror.c [hunk 1]
```diff
@@ -65,25 +66,59 @@ static inline const char *strerror_(const errinfo_t *tab, size_t tabsz, const ch
...
+
+	return ret;
 }
 
 
 const char *gai_strerror(int errnum)
 {
-	return strerror_(gaitab, sizeof(gaitab), gainames, errnum);
+	const char *str;
+	int err;
```
Comment (kemonats) [inline string/strerror.c:116]: initialization of the err variable is missing
```suggestion
	int err = 0;
```

**Comment by badochov** [comment]
I've never seen those failures on imxrt117 before however the code seems valid

**Comment by agkaminski** [comment]
> I've never seen those failures on imxrt117 before however the code seems valid

Random fail yet again

**Review by badochov — APPROVED** [review]
LGTM

**Pull request merged**